src/py42/clients/alerts.py

```diff
@@ -136,3 +136,32 @@ def get_aggregate_data(self, alert_id):
             :class:`py42.response.Py42Response`
         """
         return self._alert_service.get_aggregate_data(alert_id)
+
+    def get_all_alert_details(self, query, ascending=True):
+        """
+        Helper method that combines `.get_all_pages()` and `.get_details()`.
+        Returns an iterator of alert detail objects in chronological order by alert
+        creation date.
+
+        Args:
+            query (:class:`py42.sdk.queries.alerts.alert_query.AlertQuery`): An alert query.
+            ascending (bool): Determines if sort ordering should be ascending/descending
+                based on alert creation time (overrides the sort order on the query object).
+                Defaults to True.
+
+        Returns:
+            generator: An object that iterates over alert detail items.
+        """
+        query.page_size = 100
+        query.sort_direction = "asc" if ascending else "desc"
```

Comment:
why not simply honor the `page_size` and `sort_direction` that the user passes in on the `query`?

Reply:
`get_details()` only accepts a max of 100 ids, so we're just getting only what we can process in each iteration.